### PR TITLE
Add RARE (CRNN-seq2seq) README and support O2 training

### PR DESCRIPTION
**Pull request by zhtmike**
- Add RARE (CRNN-seq2seq) README and benchmark result. 
- The native GRUCell operator does not support FP16 input (see issue https://gitee.com/mindspore/mindspore/issues/I6ZY2O), here we implemented a input-fixed version
- Update the multi dataset evaluation script due to the interface update (num_columns -> columns_index), handle the exception when data_dir_root is incorrect

Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/models/_nn/rnn_cells.py
Comment (SamitHuang) [inline mindocr/models/_nn/rnn_cells.py:1]: 1. 可否通过修改amp.py里的white/black list来将RNN层设成半精度？
2. 建议rename  _nn -> utils，可把自定义的layers 和 helpers放里面

Comment (SamitHuang) [inline mindocr/models/_nn/rnn_cells.py:1]: 1. 可否通过修改amp.py里的white/black list来将RNN层设成半精度？
2. 建议rename  _nn -> utils，可把自定义的layers 和 helpers放里面

Comment (zhtmike) [reply mindocr/models/_nn/rnn_cells.py:1]: RNNCell应当属于white list, 会把input 自动cast为fp16。但目前mindspore RNNCell会在fp16输入时前向报错，原因是RNNCell内参如weight_ih没有自动转换成fp16，从而当用到MatMul算子时因为两边输入type不对齐而报错。此改动强制将内参和输入转换成统一type，避免这一问题。

Comment (SamitHuang) [reply mindocr/models/_nn/rnn_cells.py:1]: 了解，那应该在自定义的RNNCell init中加入 cast_fp16的选项？否则如果设成O0的话，该层仍然会半精度计算。

Comment (zhtmike) [reply mindocr/models/_nn/rnn_cells.py:1]: 现在RNNCell内参会根据输入的type改变的，如果输入为fp32，那内参会改为fp32，计算会是全精度

**Comment by zhtmike** [comment]
Uploaded checkpoint path, can review @SamitHuang @HaoyangLee

**Comment by SamitHuang** [comment]
Conflicts are to be solved before merge

**Review by HaoyangLee — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**